File: python/sorts/selection_sort.py
# Python implementation for Selection Sort
def selection_sort(arr):
	for i in range(len(arr)):
		# Find min in unsorted remainder of array
		min = i
		for j in range(i+1, len(arr)):
			if arr[min] > arr[j]:
				min = j
				
		# Swap the min with first element
		arr[i], arr[min] = arr[min], arr[i]

File: c/Sorts/SelectionSort.c
/**
 * Selection Sort
 * sorts an array of integers
 * array: the array to be sorted
 * length: the length of the array
 */ 
void selectionSort(int *array, int length) {
    int i;
    int j;

    for (i = 0; i < length; i++) {
        int minimum = i;
        for (j = i + 1; j < length; j++) {
            if (array[j] < array[minimum]) {
                minimum = j;
            } 
        }
        int temp = array[i];
        array[i] = array[minimum];
        array[minimum] = temp;
    }
}
